### ISSUE TITLE
The terminology surrounding Max Buffer Size is inconsistent

### ISSUE THREAD

**Issue by ermo** [znc/znc#1281]
Today, as I was configuring my ZNC instance using the webUI module, I noticed that the terminology around the size of the channel playback buffer is not exactly uniform.

I therefore propose that the webUI module be updated to use the following terminology and tooltips:

**"MaxBufferSize"** (used on the _ZNC » webadmin » Settings_ page)  :

> "This is the amount of lines that the playback buffer will store for channels 
> before dropping off the oldest line. The buffers are stored in the memory by default. MaxBufferSize controls the global maximum allowed amount of lines in the playback buffer for each channel."

**"BufferSize"** (used on the  _ZNC » webadmin » Edit User [user]_ page) : 

> "This is the amount of lines that the playback buffer will store for channels before dropping off the oldest line. The buffers are stored in the memory by default. For normal users, the global MaxBufferSize setting controls the global maximum allowed number of lines in the playback buffer for each channel. Admin users are not subject to this limit."

**"BufferCount"** (used on the _ZNC » webadmin » Edit User [user] » Edit Network [network]_ page) : 

Delete and replace it with **"BufferSize"** instead for consistency.

**Comment by teward** [comment]
`MaxBufferSize` is accurate and should be left alone.  It is the global maximum buffer size permitted for either the 'user' level setting or for individual channels (and only admins can go beyond it).

`BufferSize` at the user level is for user 'default' maximum.  New channel buffers for a user's network(s) are autoset to this value.  Perhaps a rename to `MaxNetworkBufferSize` or `MaxNetBufferSize`.

`BufferCount` would be a per-network 'default' for channels under that network.  As well, newly added networks inherit the default maximum.  This should be left alone, except for a rename to `MaxChanBufferSize` perhaps.

What's missing in your analysis is a buffer count at the network channel level, which I don't remember the name (for the purposes of my reply, assume it's just ChanBufferSize).

Basically, for a non ZNC-administrator user, the logic is thus:

`MaxBufferSize >= BufferSize (user level) >= BufferCount (network level) >= ChanBufferSize`

"Delete it" as you suggests is vague, though, are you saying to delete BufferCount, or overall rename everything?

**Comment by ermo** [comment]
Having a self-documenting UI is IMHO a really nice feature in terms of discoverability, so ensuring that the labels and their accompanying tooltip text (where applicable) are both accurate and succint is therefore worth striving for.

In the user facing webinterface, I would personally prefer full CamelCase terms:

MaxBufferSize (global maximum buffersize)
UserBufferSize (defaults to MaxBufferSize)
NetworkBufferSize (defaults to UserBufferSize)
ChannelBufferSize (defaults to NetworkBufferSize)

I'll let others determine whether or not the above terminology has any value in the code proper.

**Comment by cmcginty** [comment]
> MaxBufferSize is accurate and should be left alone.

It's not accurate in that the value has no unit. Had to search around to determine if this was bytes, lines, or some other unit.

**Comment by lodenrogue** [comment]
What is the definition of MaxBufferSize? global maximum buffersize doesn't tell me much.
